src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PreReleaseContactsService.cs

```diff
@@ -0,0 +1,299 @@
+                        .Include(r => r.User)
+                        .AnyAsync(r =>
+                            r.ReleaseId == releaseId
+                            && r.User.Email.ToLower() == email.ToLower()
+                            && r.Role == ReleaseRole.PrereleaseViewer))
+                    {
+                        var existingUser = await _context
+                            .Users
+                            .Where(u => u.Email.ToLower() == email.ToLower())
+                            .FirstOrDefaultAsync();
+
+                        if (existingUser != null)
+                        {
+                            _context.Add(new UserReleaseRole
+                            {
+                                ReleaseId = releaseId,
+                                Role = ReleaseRole.PrereleaseViewer,
+                                UserId = existingUser.Id
+                            });
+                            await _context.SaveChangesAsync();
+                        }
+                        else
+                        {
+                            var existingInvite = await _context
+                                .UserReleaseInvites
+                                .Where(i =>
+                                    i.ReleaseId == releaseId
+                                    && i.Email.ToLower() == email.ToLower()
+                                    && i.Role == ReleaseRole.PrereleaseViewer)
+                                .FirstOrDefaultAsync();
+
+                            if (existingInvite == null)
+                            {
+                                newUser = true;
+
+                                _context.Add(new UserReleaseInvite
+                                {
+                                    Email = email.ToLower(),
+                                    ReleaseId = releaseId,
+                                    Role = ReleaseRole.PrereleaseViewer,
+                                    Created = Now,
+                                    CreatedById = _userService.GetUserId()
+                                });
+                                await _context.SaveChangesAsync();
+
+                                var existingInviteToSystem = await _usersAndRolesDbContext
+                                    .UserInvites
+                                    .Where(i => i.Email.ToLower() == email.ToLower())
+                                    .FirstOrDefaultAsync();
+
+                                // TODO EES-1181 - allow multiple invites per email address to allow people to 
+                                // be assigned multiple roles upon first login
+                                if (existingInviteToSystem == null)
+                                {
+                                    var prereleaseRole = await _usersAndRolesDbContext
+                                        .Roles
+                                        // TODO represent Roles with an Enum
```

Comment:
I think it's worth doing this pretty soon if we don't already have a ticket for it and then we can clean up things like the Role names in UsersAndRolesDbContext.

Reply:
Created https://alpha-dfe-data.atlassian.net/browse/EES-1257 to cover this